setup.py

```diff
@@ -51,11 +66,6 @@ def is_installing():
         module_check_fn = _VERSION_GLOBALS['_check_module_dependencies']
         module_check_fn(is_nilearn_installing=True)
 
-    install_requires = \
-        ['%s>=%s' % (mod, meta['min_version'])
-            for mod, meta in _VERSION_GLOBALS['REQUIRED_MODULE_METADATA']
-            if not meta['required_at_installation']]
```

Comment:
@GaelVaroquaux 
This line was filtering out packages which were marked as required: True in `version.py`. I don't understand why this is here at all, and it concerns me that it is so much work to understand this.
Can you tell me why this line is here?
We document &/or streamline the setup files.
cc: @jeromedockes